pages/index.tsx

```diff
@@ -19,21 +34,136 @@ export default function Home(): ReactElement {
     </Container>
   )
 }
 
+export interface QueryParams {
+  user_key: string
+  uniqueNumber: string
+}
+
+function buildApiUrl(url: string, queryParams: QueryParams) {
+  const apiUrl = new URL(url)
+
+  for (const key in queryParams) {
+    apiUrl.searchParams.append(key, queryParams[key as 'user_key' | 'uniqueNumber'])
+  }
+
+  return apiUrl.toString()
+}
+
 export const getServerSideProps: GetServerSideProps = async ({ req, locale }) => {
   const isProd = process.env.NODE_ENV === 'production'
   const logger = isProd ? pino({}) : pino({ prettyPrint: true })
   logger.info(req)
 
+  /*
+   * Load environment variables to be used for authentication & API calls.
+   * @TODO: Handle error case where env vars are null or undefined.
+   */
+  // Request fields
+  const ID_HEADER_NAME: string = process.env.ID_HEADER_NAME ?? ''
+
+  // API fields
+  const API_URL: string = process.env.API_URL ?? ''
+  const API_USER_KEY: string = process.env.API_USER_KEY ?? ''
+
+  // TLS Certificate fields
+  const CERT_DIR: string = process.env.CERTIFICATE_DIR ?? ''
+  const P12_FILE: string = process.env.P12_FILE ?? ''
+  const P12_PATH: string = path.join(CERT_DIR, P12_FILE)
+
+  let apiData: JSON | null = null
+
+  // Returns certificate object with cert, key, and ca fields.
+  // https://dexus.github.io/pem/jsdoc/module-pem.html#.readPkcs12
+  async function getCertificate() {
+    const pemReadPkcs12 = promisify(pem.readPkcs12)
+    const pfx = fs.readFileSync(P12_PATH)
+
+    // TS does not play very nicely with util.promisify
+    // See, e.g., https://github.com/Microsoft/TypeScript/issues/26048
+    // Non-MVP TODO: Consider removing this ignore & TypeScriptifying.
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore -- TypeScript does not handle promisify well.
+    const keybundle = await pemReadPkcs12(pfx, {})
+    return keybundle
+  }
+
+  // Takes certificate that getCertificate function returns as argument,
+  // makes API call, returns all API data.
+  async function makeRequest(certificate: Pkcs12ReadResult) {
+    const headers = {
+      Accept: 'application/json',
+    }
+
+    /* https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options
+
+    // TODO: Store certs in memory, instead of loading every time.
+    // TODO: rejectUnauthorized - set to true or implement a `checkServerIdentity`
+    // function to check that the certificate is actually
+    //  issued by the host you're connecting to.
+    //  https://nodejs.org/api/https.html#https_https_request_url_options_callback
+    */
+    const options = {
+      cert: certificate.cert,
+      key: certificate.key,
+      rejectUnauthorized: false,
+      keepAlive: false,
+    }
+
+    // Instantiate agent to use with TLS Certificate.
+    // Reference: https://sebtrif.xyz/blog/2019-10-03-client-side-ssl-in-node-js-with-fetch/
+    const sslConfiguredAgent: https.Agent = new https.Agent(options)
+
+    // TODO: if no uniqueNumber, redirect.
```

Comment:
Let's ticket this? If there's no unique number and this is being accessed via the user-facing claim status url, this should be a Sev 3 alert.

Reply:
Good call. I've created #217, so I'll remove this TODO